index.js

```diff
@@ -0,0 +1,68 @@
+const validateSingleTeamRule = teamIds => (
+  Object.values(teamIds).reduce((valid, teamCount) => (
+    valid && teamCount <= 2
+  ), true)
+)
+
+const validateSingleGameRule = gameIds => (
+  Object.values(gameIds).reduce((valid, gameCount) => (
+    valid && gameCount <= 3
+  ), true)
+)
+
+const validatePositionCount = positions => (
+  Object.values(positions).reduce((valid, position) => (
+    valid && position === 0
+  ), true)
+)
+
+const validateSalaryRule = salary => salary <= 45000
+
+
+const validateLineup = (lineup) => {
+  const teamIds = {}
+  const gameIds = {}
+  let totalSalary = 0
+  const positions = {
+    P: 1,
+    C: 1,
+    '1B': 1,
+    '2B': 1,
+    '3B': 1,
+    SS: 1,
+    OF: 3
+  }
+
+  lineup.forEach(({ teamId, gameId, salary, position }) => {
```

Comment:
Nice use of destructuring in the function call, you see that a lot in React code.